Create pipeline which uses Terraform to create GCP resources (#77)
bakdata/ci-templates: .github/workflows/README.md

    steps:
      - run: echo Bumped Version from ${{ needs.call-workflow-passing-data.outputs.old-version }} to ${{ needs.call-workflow-passing-data.outputs.release-version }}
```

## Terraform Create GKE Cluster

> This workflow leverages your Terraform project to establish a GKE cluster.

Reviewer: This is just a generic workflow for any Terraform project and could be used for something else than just a GKE cluster, right? Please adjust the name and description accordingly

### Prerequisites

> Establish a service account and save its Key as a GitHub secret so you can provide it to the variable `google-credentials`.

Reviewer: ```suggestion
Create a service account and save its key as a GitHub secret so you can provide it to the variable `google-credentials`.
```

### Input Parameters

| Name              | Required | Default Value |  Type  | Description                                  |
| ----------------- | :------: | :-----------: | :----: | -------------------------------------------- |
| tf-plan           |    ❌    |    tfplan     | string | Name of the terraform plan output file       |
| working-directory |    ❌    |       .       | string | Directory containing your Terraform project. |

### Secret Parameters

> A service account (SA) with the necessary permissions is necessary for creating a GKE cluster. Generate a key out of the recently created SA and provide it to the workflow.

Reviewer: ```suggestion
A service account (SA) with the necessary permissions is required for creating a GKE cluster. Generate a key for the recently created SA and provide it to the workflow.
```

| Name               | Required | Description                       |
| ------------------ | :------: | --------------------------------- |
| google-credentials |    ✅    | Key of the service account to use |

### Calling the workflow

```yaml

> name: Release

Reviewer: ```suggestion
name: Terraform
```

on:
  workflow_dispatch:

jobs:
  call-workflow-passing-data:

>     name: Terraform Create GKE Cluster

Author: ups..

>     uses: bakdata/ci-templates/.github/workflows/tf-gke-cluster.yaml@main

Reviewer: The example also mentions cluster still
